Content.Server/GameObjects/EntitySystems/GravitySystem.cs

```diff
@@ -0,0 +1,44 @@
+using System;
+using System.Collections.Generic;
+using Content.Server.GameObjects.Components.Gravity;
+using JetBrains.Annotations;
+using Robust.Shared.GameObjects;
+using Robust.Shared.GameObjects.Systems;
+using Robust.Shared.Interfaces.Map;
+using Robust.Shared.IoC;
+using Robust.Shared.Map;
+
+namespace Content.Server.GameObjects.EntitySystems
+{
+    [UsedImplicitly]
+    public class GravitySystem: EntitySystem
+    {
+#pragma warning disable 649
+        [Dependency] private readonly IMapManager _mapManager;
+#pragma warning restore 649
+
+        public override void Initialize()
+        {
+            EntityQuery = new TypeEntityQuery<GravityGeneratorComponent>();
+        }
+
+        public override void Update(float frameTime)
+        {
+            var gridsWithGravity = new List<GridId>();
+            foreach (var entity in RelevantEntities)
+            {
+                var generator = entity.GetComponent<GravityGeneratorComponent>();
+                generator.UpdateSprite();
+                if (generator.Status == GravityGeneratorStatus.On)
+                {
+                    gridsWithGravity.Add(entity.Transform.GridID);
+                }
+            }
+
+            foreach (var grid in _mapManager.GetAllGrids())
+            {
+                grid.HasGravity = gridsWithGravity.Contains(grid.Index);
+            }
```

Comment:
I think you should try iterating the list of grids with gravity instead of iterating all grids

Reply:
The reason why I'm iterating the whole list of grids here is so that grids that are registered as having gravity but then lose it are updated properly.

Reply:
Ah, that makes sense now--
Yeah, then this is the way to go about that.